packages/server/lib/util/resolve.js

```diff
@@ -21,7 +23,10 @@ module.exports = {
 
       debug('resolving typescript with options %o', options)
 
-      const resolved = resolve.sync('typescript', options)
+      // use built-in resolve when yarn v2 is detected
+      const resolved = fs.existsSync(path.join(projectRoot, '.yarn', 'cache')) ?
+        require.resolve('typescript') :
+        resolve.sync('typescript', options)
```

Comment:
We could also just fall back to the more extensible/flexible `require.resolve` if the `resolve.sync` result is falsy. Not sure which is preferred here
```suggestion
      const resolved = resolve.sync('typescript', options) || require.resolve('typescript')        
```